packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -723,12 +722,6 @@ export function forceRehydration(vm: VM) {
     }
 }
 
-export function isLightRenderModeVM(vm: VM): vm is VM & { cmpRoot: ShadowRoot } {
-    // We don't refer to vm.def.ctor.shadow because that could be changed by user
-    // after instantiation.
-    return !isNull(vm.cmpRoot);
-}
-
 function getRenderRoot(vm: VM) {
-    return isLightRenderModeVM(vm) ? vm.cmpRoot : vm.elm;
+    return vm.cmpRoot ?? vm.elm;
```

Comment:
This would avoid issues where the VM enters into a wrong state.

```suggestion
    return vm.renderMode === RenderMode.Shadow ? vm.cmpRoot : vm.elm;
```